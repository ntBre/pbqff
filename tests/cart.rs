use std::{fs::read_to_string, path::Path};

use assert_cmd::Command;
use insta::{assert_snapshot, with_settings};
use tempfile::tempdir;

#[test]
fn cart() -> std::io::Result<()> {
    let config_file = Path::new("testfiles/cart.toml");
    let dir = tempdir()?;
    std::fs::copy(config_file, dir.path().join("cart.toml"))?;
    let mut cmd = Command::cargo_bin("pbqff").unwrap();
    let assert = cmd.arg("cart.toml").current_dir(&dir).assert();
    let output = assert.get_output();

    assert!(
        output.status.success(),
        "stderr: {}\nlog: {}",
        String::from_utf8_lossy(&output.stderr),
        read_to_string(dir.path().join("pbqff.log"))?,
    );

    // filter out essentially all of the numerical results just to check that
    // all of the sections are there
    with_settings!(
        {filters => vec![
            (r"(?m)^PID: \d+$", "PID: [PID]"),
            (r"(?s)(normalized geometry:).*(point group)", "$1[Normalized Geometry]\n$2"),
            (r"(?ms)^STATE NO.*?^$", "[Vibrational States]"),
            (r"(?s)(Transformed Geometry \(Ã…\):).*(Equilibrium)", "$1\n[Transformed Geometry]\n$2"),
            (r"(?ms)(Equilibrium Rotational Constants \(cm-1\):).*?^$", "$1\n[Rotational Constants]\n"),
            (r"(?s)(Geometry:).*(Vibrational)", "$1\n[Geometry]\n$2"),
            (r"(?s)(Vibrational Frequencies \(cm-1\):).*?(ZPT)", "$1\n[Freqs]\n$2"),
            (r"(?ms)(^Rotational Constants \(cm-1\):).*?(Coriolis Resonances)", "$1\n[Rotational Constants]\n$2"),
        ]}, {
            assert_snapshot!(read_to_string(dir.path().join("pbqff.out")).unwrap());
        }
    );

    Ok(())
}
