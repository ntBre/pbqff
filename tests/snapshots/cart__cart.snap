---
source: tests/cart.rs
expression: "read_to_string(dir.path().join(\"pbqff.out\")).unwrap()"
snapshot_kind: text
---
PID: [PID]
version: deadbeef
normalized geometry:[Normalized Geometry]
point group:C2v(Axis(Y), Plane(Axis(Y), Axis(Z)), Plane(Axis(X), Axis(Y)))
15 Cartesian coordinates requires 11952 points
[Vibrational States]
Transformed Geometry (Ã…):
[Transformed Geometry]
Equilibrium Rotational Constants (cm-1):
[Rotational Constants]

Geometry:
[Geometry]
Vibrational Frequencies (cm-1):
[Freqs]
ZPT =   6675.1

Rotational Constants (cm-1):
[Rotational Constants]
Coriolis Resonances:
    4    3    C
    5    4    A
    6    5    A
    7    6    B
    8    5    B
    8    7    A

Type 1 Fermi Resonances:
    5    2
    6    2
    7    2
    8    2

Type 2 Fermi Resonances:
    4    2    1
    6    2    1
    6    4    2
    8    2    0

normal termination of pbqff
